### PR TITLE
chore: upgrade geth and nethermind for sim tests

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Test compatibility with latest versions of all clients. 

**Description**

Upgrade **Geth** and **Nethermind** for the sim tests. 

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

**Review by jeluard — APPROVED** [review]

**Comment by jeluard** [comment]
Would it make sense to store those values so that they are reused by `e2e` tests too? Right now they are also defined in `test.yml` and `CONTRIBUTING.md`. 
It also forces users to provide those value manually when running `scripts/run_e2e_env.sh`.

File: .github/workflows/test-sim.yml [hunk 1]
```diff
@@ -23,9 +23,9 @@ on:
         default: 40
 
 env:
-  GETH_DOCKER_IMAGE: ethereum/client-go:v1.11.6
+  GETH_DOCKER_IMAGE: ethereum/client-go:v1.13.11
   LIGHTHOUSE_DOCKER_IMAGE: sigp/lighthouse:latest-amd64-modern-dev
```
Comment (nflaig) [inline .github/workflows/test-sim.yml:27]: we should use a release version for lighthouse as well

Comment (nazarhussain) [reply .github/workflows/test-sim.yml:27]: Lighthosue releases does not build with `minimal` preset.

Comment (nflaig) [reply .github/workflows/test-sim.yml:27]: what about [sigp/lighthouse:v4.6.0-amd64-modern-dev](https://hub.docker.com/layers/sigp/lighthouse/v4.6.0-amd64-modern-dev/images/sha256-21b34717f5ebbbda4736e08aadf8a6784f06b7fe6686337411a29f4c3414f673?context=explore), the `-dev` suffix suggest it has minimal preset.

**Comment by nflaig** [comment]
> Would it make sense to store those values so that they are reused by `e2e` tests too? Right now they are also defined in `test.yml` and `CONTRIBUTING.md`. It also forces users to provide those value manually when running `scripts/run_e2e_env.sh`.

I think that is a good idea, we should make as easy as possible to run tests locally, also this PR did not update value in `CONTRIBUTING.md`, which seems incomplete to me

File: .github/workflows/test-sim.yml (hunk 1 above)
Comment (nazarhussain) [reply .github/workflows/test-sim.yml:27]: Yes the build with `-dev` is the right one. Seems for latest releases they started the dev builds. I will update it shortly.

**Comment by nazarhussain** [comment]
> reused by `e2e` tests too

Reusability is always great. Will try to find better approach to share across scripts, workflows and local env. Probably an env file will work out. Will double check.

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.16.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.16.0) :tada:

**Pull request merged**